### PR TITLE
[DO NOT MERGE] Initial work in support of the AUAV ESC35 board

### PR DESCRIPTION
**Pull request by Inspirati**

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@davids5 How far off from being mergable is this?

@Inspirati What is the plan with this hardware - are you ready to spin up efforts?

**Comment by Inspirati** [comment]
Thanks to David completing the bootloader work in this PR and having merged his results in branch nuttx_v3.

I'm continuing with development on the application code side.

**Comment by pkocmoud** [comment]
I have tested it and the bootloader works fine using a PixRacer.

File: nuttx-configs/esc35-v1/include/board.h [hunk 1]
```diff
@@ -121,8 +121,8 @@
 
 #define STM32_CFGR_USBPRE       0
 
-#define GPIO_USART2_RX         GPIO_USART2_RX_2
```
Comment (davids5) [inline nuttx-configs/esc35-v1/include/board.h]: @Inspirati Is this for your debug? If so and it is not pinned out on the real HW - it does not belong in the released code

**Comment by davids5** [comment]
@LorenzMeier I am alittle confused. I used @Inspirati work to make a working bootloaded for ESC3.5 in nuttx_v3, any future work should be done from nuttx_v3 not this branch.

File: src/drivers/bootloaders/src/sched/timer.c [hunk 2]
```diff
@@ -321,7 +321,7 @@ bl_timer_id timer_allocate(bl_timer_modes_t mode, time_ms_t msfromnow,
```
Comment (davids5) [inline src/drivers/bootloaders/src/sched/timer.c]: @Inspirati  Please branch nuttx_v3 and Cherry pick this commit in to you new branch.

File: nuttx-configs/esc35-v1/bootloader/defconfig [hunk 3]
```diff
@@ -296,9 +296,9 @@ CONFIG_ARCH_CHIP_STM32F303CC=y
 # CONFIG_STM32_FLASH_CONFIG_6 is not set
 # CONFIG_STM32_FLASH_CONFIG_8 is not set
 # CONFIG_STM32_FLASH_CONFIG_B is not set
-# CONFIG_STM32_FLASH_CONFIG_C is not set
+# CONFIG_STM32_FLASH_CONFIG_C=y
```
Comment (davids5) [inline nuttx-configs/esc35-v1/bootloader/defconfig]: @Inspirati 

This looks incorrect to me. I submitted the whole CONFIG_STM32_FLASH_CONFIG stuff upstream to Greg a long while back to handle F4 series parts that have non regular flash page sizes. The F3 has homogeneous 2Kib flahs pages and I believe this does not work as expected 

SIZE=N*pagesize on an F3 works
SIZE=N*pagesize on an F4 is not defined.

**Comment by LorenzMeier** [comment]
I'm closing this since it seems inactive. We do have new NuttX now on master.

**Pull request closed without merging**